endstate_correction/protocol.py

```diff
@@ -296,39 +303,39 @@ def perform_endstate_correction(protocol: Protocol) -> Results:
         elif (
             protocol.direction == "unidirectional_forward"
             or protocol.direction == "unidirectional"
-            ):
+        ):
             # perform switching from mm to qml
             print("Performing unidirectional protocol from MM to QML...")
             lambs = np.linspace(0, 1, protocol.neq_switching_length)
             Ws, endstates_mm_to_qml, trajs_mm_to_qml = perform_switching(
-                    sim,
-                    lambs,
-                    samples=protocol.trajectories[0],
-                    nr_of_switches=protocol.nr_of_switches,
-                    save_endstates=protocol.save_endstates,
-                    save_trajs=protocol.save_trajs
-                )
-            Ws_from_mm_to_qml = np.array(Ws/ kBT)
+                sim,
+                lambs,
+                samples=protocol.trajectories[0],
+                nr_of_switches=protocol.nr_of_switches,
+                save_endstates=protocol.save_endstates,
+                save_trajs=protocol.save_trajs,
+            )
+            Ws_from_mm_to_qml = np.array(Ws / kBT)
 
             r.W_mm_to_qml = Ws_from_mm_to_qml
             r.endstate_samples_mm_to_qml = endstates_mm_to_qml
             r.switching_traj_mm_to_qml = trajs_mm_to_qml
 
-        elif (protocol.direction == "unidirectional_reverse"):
+        elif protocol.direction == "unidirectional_reverse":
             # perform switching from qml to mm
             lambs = np.linspace(1, 0, protocol.neq_switching_length)
             Ws, endstates_qml_to_mm, trajs_qml_to_mm = perform_switching(
-                    sim,
-                    lambs,
-                    samples=protocol.trajectories[-1],
-                    nr_of_switches=protocol.nr_of_switches,
-                    save_endstates=protocol.save_endstates,
-                    save_trajs=protocol.save_trajs
-                )
-            Ws_from_qml_to_mm = np.array(Ws/ kBT)   
+                sim,
+                lambs,
+                samples=protocol.trajectories[1],
```

Comment:
Same here. -1 makes sense as one is taking the last frame but I don't understand why you need to take the second frame

Reply:
has been resolved in https://github.com/wiederm/endstate_correction/commit/ff080e2a50b5b2982b805c8d246ca9bf753c4edb